File: KoKoslan/src/java/ast/KoKoMINUS.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.ast;
import java.util.*;
import java.io.*;


public class KoKoMINUS extends KoKoBiOperation {
	
    public KoKoMINUS(KoKoAst operator, KoKoAst left, KoKoAst right){
		super(operator, left, right);
	}
	
	@Override
	public KoKoValue eval(KoKoContext ctx){
	   try {
		     KoKoNumValue lv = (KoKoNumValue)(left().eval(ctx));
			 KoKoNumValue rv = (KoKoNumValue)(right().eval(ctx));
	         return new KoKoNumValue(lv.getValue() - rv.getValue());
	   } catch (Exception e) {
			throw new KoKoEvalException(e.getMessage());
	   }
	   
   }
	
}

File: KoKoslan/src/kotlin/ast/KoKoMinus.kt
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.ast
import java.util.*
import java.io.*

class KoKoMINUS(Operator : KoKoAst, left : KoKoAst, right : KoKoAst) : KoKoBiOperation(Operator, left, right) {
	
	override fun eval(ctx : KoKoContext) : KoKoValue{
	   try {
		     val lv = (KoKoNumValue)(left().eval(ctx))
			 val rv = (KoKoNumValue)(right().eval(ctx))
	         return KoKoNumValue(lv.getValue() - rv.getValue())
	   } catch (e : Exception) {
			throw KoKoEvalException(e.getMessage())
	   }
	   
   }
	
}